components/nvs_flash/src/nvs_page.cpp

```diff
@@ -48,16 +48,16 @@ esp_err_t Page::load(uint32_t sectorNumber)
         mState = header.mState;
         // check if the whole page is really empty
         // reading the whole page takes ~40 times less than erasing it
-        uint32_t line[8];
+        uint32_t line[64];
```

Comment:
Hi Gautier,

The only concern I have with this is that allocating 512 bytes on the stack (rather than 32) is quite a lot of stack, and some tasks may not have this spare.

A few possible ideas:

* Do you have any performance numbers on smaller size reads (bigger than 64 bytes, smaller than 512)?
* I suspect it might be possible to use spi_flash_mmap() the whole sector instead, provided flash encryption is disabled. This would probably be the fastest, although it needs two code paths (if flash encryption is enabled, spi_flash_read() will still be required)
* Could simply allocate the `line` buffer from heap, I guess.

I've asked @igrr  and @mahavirj to review as well, as they're more familiar with NVS than I am and might have different thoughts on this.

Reply:
Hi Angus,

 - I didn't spend benchmark various values so far.
 - Yes, I was thinking about using spi_flash_mmap() as well, as it would be a single action, and no stack usage. Performance would be related to the size of a cache line.
 - correct, heap could be used here.

I agree, NVS maintainers could likely provide the best fix.